osu.Server.Spectator/Hubs/MultiplayerHubContext.cs

```diff
@@ -128,26 +128,27 @@ public async Task ChangeAndBroadcastUserState(ServerMultiplayerRoom room, Multip
         {
             log(room, user, $"User state changed from {user.State} to {state}");
 
-            user.State = state;
-
             string? connectionId = users.GetConnectionIdForUser(user.UserID);
 
             if (connectionId != null)
             {
-                switch (state)
+                if (state == MultiplayerUserState.FinishedPlay)
                 {
-                    case MultiplayerUserState.FinishedPlay:
-                    case MultiplayerUserState.Idle:
-                        await context.Groups.RemoveFromGroupAsync(connectionId, MultiplayerHub.GetGroupId(room.RoomID, true));
-                        break;
-
-                    case MultiplayerUserState.Ready:
-                    case MultiplayerUserState.Spectating:
+                    // when entering FinishedPlay state, we can be confident that player can be removed from group
+                    await context.Groups.RemoveFromGroupAsync(connectionId, MultiplayerHub.GetGroupId(room.RoomID, true));
+                }
+                else
+                {
+                    // when leaving Results state, we can be confident that player can be removed from group
```

Comment:
Huh?

Reply:
Oops, comment meant to say "can be added to group". Nevertheless, I will remove it.